src/cli/msig/index.ts

```diff
@@ -164,6 +166,20 @@ export default async function msig(args: string[]) {
             );
 
             if (test) await executeProposal(proposer, proposalName, proposalHash);
+        } else if (proposalType === 'deploy-contract') {
+            const contractName = args[3];
+
+            if (!contractName) {
+                throw new Error('Contract name must be provided for deploy-contract proposal');
+            }
+
+            const contractInfo = {
+                account: contractName,
+                contractDir: path.join(__dirname, `../../Tonomy-Contracts/contracts/${contractName}`),
+            };
+
+            await deployContract(contractInfo, signer, { throughTonomyProxy: true });
```

Comment:
this needs to be done as a proposal with `createProposal()` and `if (test) executeProposal()`